server/apps/archive/archive_rewrite.py

```diff
@@ -0,0 +1,138 @@
+
+logger = logging.getLogger(__name__)
+
+
+class ArchiveRewriteResource(Resource):
+    endpoint_name = 'archive_rewrite'
+    resource_title = endpoint_name
+
+    schema = {
+        'desk_id': {"type": "string"}
+    }
+
+    url = 'archive/<{0}:original_id>/rewrite'.format(item_url)
+
+    resource_methods = ['POST']
+    privileges = {'POST': 'rewrite'}
+
+
+class ArchiveRewriteService(Service):
+    def create(self, docs, **kwargs):
+        original_id = request.view_args['original_id']
+        doc = docs[0]
+        desk_id = doc.get('desk_id')
+        archive_service = get_resource_service('archive')
+        original = archive_service.find_one(req=None, _id=original_id)
+        digital = TakesPackageService().get_take_package(original)
+
+        self._validate_rewrite(original)
+        rewrite = self._create_rewrite_article(original, digital, desk_id)
+        archive_service.post([rewrite])
+        build_custom_hateoas(CUSTOM_HATEOAS, rewrite)
+        self._add_rewritten_flag(original, digital, rewrite)
+        return [rewrite]
+
+    def _validate_rewrite(self, original):
+        """
+        Validates the article to be rewritten
+        :param original: article to be rewritten
+        :raises: SuperdeskApiError
+        """
+        if not original:
+            raise SuperdeskApiError.notFoundError(message='Cannot find the article')
+
+        if not original.get('event_id'):
+            raise SuperdeskApiError.notFoundError(message='Event id does not exist')
+
+        if get_resource_service('published').is_rewritten_before(original['_id']):
+            raise SuperdeskApiError.badRequestError(message='Article has been rewritten before !')
+
+        if not is_workflow_state_transition_valid('rewrite', original['state']):
+            raise InvalidStateTransitionError()
+
+        if not TakesPackageService().is_last_takes_package_item(original):
+            raise SuperdeskApiError.badRequestError(message="Only last take of the package can be rewritten.")
+
+    def _create_rewrite_article(self, original, digital, desk_id):
+        """
+        Creates a new story and sets the metadata from original and digital
+        :param original: original story
+        :param digital: digital version of the story
+        :param desk_id: desk it belongs to
+        :return:new story
+        """
+        rewrite = dict()
+        fields = ['family_id', 'abstract', 'anpa_category',
+                  'pubstatus', 'slugline', 'urgency', 'subject', 'dateline',
+                  'priority', 'byline', 'dateline', 'headline', 'event_id']
+
+        for field in fields:
+            if original.get(field):
+                rewrite[field] = original[field]
+
+        if digital:
+            # check if there's digital
+            rewrite['rewrite_of'] = digital['_id']
+        else:
+            # if not use original's id
+            rewrite['rewrite_of'] = original['_id']
+
+        rewrite['task'] = original['task']
```

Comment:
`desk_id` is unused and the `user_id` in `original[task][user]` and user who is rewriting can be different. Also, the rewritten article should go to incoming stage.